### PR TITLE
Restructure CI manifests, document CI Jobs

### PR DESCRIPTION
**Pull request by afrittoli**
Move CI manifests around to clearly separate resources:
- trigger template for all repos
- trigger template for plumbing
- task, ci jobs and conditions

Define a fixed set of parameters that trigger templates
may rely on from the event listener + interceptors +
overlays + bindings, even if this means certain params
may not be used by some CI jobs.

Change the PR kind label check to align to other checks
so move it from a TaskRun to a PipelineRun and move the
pipeline definition to its own manifest.

Document the trigger template input interface
and the process for adding a new CI job.

Signed-off-by: Andrea Frittoli <andrea.frittoli@uk.ibm.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

/kind documentation

### CODE REVIEW

File: tekton/ci/jobs/tasks.yaml [hunk 1]
```diff
@@ -200,4 +138,89 @@ spec:
   volumes:
   - name: gcp-secret
     secret:
-      secretName: release-secret
\ No newline at end of file
+      secretName: release-secret
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
```
Comment (afrittoli) [inline tekton/ci/jobs/tasks.yaml:144]: This was extracted with no change from `check-pr-has-kind-label.yaml `

Comment (afrittoli) [inline tekton/ci/jobs/tasks.yaml:144]: This was extracted with no change from `check-pr-has-kind-label.yaml `

**Comment by sbwsg** [comment]
oops, did not spot the WIP until I approved.

/approve cancel

**Review by vdemeester — APPROVED** [review]
/meow

**Comment by dibyom** [comment]
Nice!!
/lgtm

**Pull request merged**